launcher/modplatform/atlauncher/ATLPackInstallTask.cpp

```diff
@@ -337,58 +335,33 @@ QString PackInstallTask::getDirForModType(ModType type, QString raw)
 
 QString PackInstallTask::getVersionForLoader(QString uid)
 {
-    if (m_version.loader.recommended || m_version.loader.latest || m_version.loader.choose) {
-        auto vlist = APPLICATION->metadataIndex()->get(uid);
-        if (!vlist) {
-            emitFailed(tr("Failed to get local metadata index for %1").arg(uid));
-            return Q_NULLPTR;
-        }
-
-        vlist->waitToLoad();
-
-        if (m_version.loader.recommended || m_version.loader.latest) {
-            for (int i = 0; i < vlist->versions().size(); i++) {
-                auto version = vlist->versions().at(i);
-                auto reqs = version->requiredSet();
-
-                // filter by minecraft version, if the loader depends on a certain version.
-                // not all mod loaders depend on a given Minecraft version, so we won't do this
-                // filtering for those loaders.
-                if (m_version.loader.type != "fabric") {
-                    auto iter = std::find_if(reqs.begin(), reqs.end(), [](const Meta::Require& req) { return req.uid == "net.minecraft"; });
-                    if (iter == reqs.end())
-                        continue;
-                    if (iter->equalsVersion != m_version.minecraft)
-                        continue;
-                }
-
-                if (m_version.loader.recommended) {
-                    // first recommended build we find, we use.
-                    if (!version->isRecommended())
-                        continue;
-                }
-
-                return version->descriptor();
-            }
-
-            emitFailed(tr("Failed to find version for %1 loader").arg(m_version.loader.type));
-            return Q_NULLPTR;
-        } else if (m_version.loader.choose) {
-            // Fabric Loader doesn't depend on a given Minecraft version.
-            if (m_version.loader.type == "fabric") {
-                return m_support->chooseVersion(vlist, Q_NULLPTR);
-            }
-
-            return m_support->chooseVersion(vlist, m_version.minecraft);
-        }
+    QString loaderVersion;
+    if (m_version.loader.recommended) {
+        loaderVersion = "recommended";
+    } else if (m_version.loader.latest) {
+        loaderVersion = "latest";
+    } else if (m_version.loader.choose) {
+        loaderVersion = "choose";
```

Comment:
```suggestion
```

Reply:
Not gonna happen: some atlauncher packs(not know any) require the user to select the version of the loader